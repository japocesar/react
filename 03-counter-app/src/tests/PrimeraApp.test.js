import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';

import PrimeraApp from '../PrimeraApp';

describe('Pruebas en PrimeraApp', () => {
    // test('Debe mostrar el mensaje "Hola, soy Goku"', () => {
    //     const saludo = 'Hola, soy Goku';
    //     const { getByText } = render( <PrimeraApp saludo={saludo} /> );
    //     // expect( getByText(saludo) ).toBeInTheDocument();
    // });

    test('debe mostrar <PrimeraAPP /> correctamente', () => {
        const saludo = 'Hola, mi nombre es Goku';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar el subtitulo enviado por props', () => {
        const saludo = 'Hola, mi nombre es Goku';
        const subtitulo = 'Test subtitle';
        const wrapper = shallow(
            <PrimeraApp
                saludo={saludo}
                subtitulo={subtitulo}
            />
        );

        const textoParrafo = wrapper.find('p').text();
        expect(textoParrafo).toBe( 'subtitulo' );
    });
    
    
});