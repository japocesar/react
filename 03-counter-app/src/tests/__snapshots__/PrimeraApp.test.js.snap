// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp debe mostrar <PrimeraAPP /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, mi nombre es Goku
  </h1>
  <p>
    Subtitulo
  </p>
</Fragment>
`;
